cluster-autoscaler/core/scale_down.go

```diff
@@ -779,45 +810,53 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
 
 			// Check how long the node was underutilized.
 			if ready && !val.Add(sd.context.ScaleDownUnneededTime).Before(currentTime) {
+				sd.addUnremovableNodeReasonSimple(node, simulator.NotUnneededLongEnough)
 				continue
 			}
 
 			// Unready nodes may be deleted after a different time than underutilized nodes.
 			if !ready && !val.Add(sd.context.ScaleDownUnreadyTime).Before(currentTime) {
+				sd.addUnremovableNodeReasonSimple(node, simulator.NotUnreadyLongEnough)
 				continue
 			}
 
 			nodeGroup, err := sd.context.CloudProvider.NodeGroupForNode(node)
 			if err != nil {
 				klog.Errorf("Error while checking node group for %s: %v", node.Name, err)
+				sd.addUnremovableNodeReasonSimple(node, simulator.UnexpectedError)
 				continue
 			}
 			if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
 				klog.V(4).Infof("Skipping %s - no node group config", node.Name)
+				sd.addUnremovableNodeReasonSimple(node, simulator.UnexpectedError)
```

Comment:
This is expected to happen for all nodes from non-autoscaled groups, so I'm not really happy about mixing it up with actual errors. Let's either skip it or use a specific reason.

Reply:
Changed to a specific reason.